trackintel/visualization/plotting.py

```diff
@@ -0,0 +1,465 @@
+    triplegs : GeoDataFrame (as trackintel triplegs)
+    locations : GeoDataFrame (as trackintel locations)
+
+    Returns
+    -------
+    tuple of floats
+        tuple with values for (north, south, east, west)
+    """
+    assert positionfixes is not None or staypoints is not None or triplegs is not None or locations is not None
+    # TODO: maybe a relative value instead of 0.03
+    if positionfixes is not None:
+        north = positionfixes.geometry.y.max()
+        south = positionfixes.geometry.y.min()
+        east = positionfixes.geometry.x.max()
+        west = positionfixes.geometry.x.min()
+    elif staypoints is not None:
+        north = staypoints.geometry.y.max() + 0.03
+        south = staypoints.geometry.y.min() - 0.03
+        east = staypoints.geometry.x.max() + 0.03
+        west = staypoints.geometry.x.min() - 0.03
+    elif triplegs is not None:
+        triplegs_bounds = triplegs.bounds
+        north = max(triplegs_bounds.maxy) + 0.03
+        south = min(triplegs_bounds.miny) - 0.03
+        east = max(triplegs_bounds.maxx) + 0.03
+        west = min(triplegs_bounds.minx) - 0.03
+    else:  # locations is not None
+        north = locations.geometry.y.max() + 0.03
+        south = locations.geometry.y.min() - 0.03
+        east = locations.geometry.x.max() + 0.03
+        west = locations.geometry.x.min() - 0.03
+    return (north, south, east, west)
+
+
+def _plot_frames(positionfixes, staypoints, triplegs, locations, radius_sp, radius_locs, ax):
+    """Plot frames on axis
+
+    Parameters
+    ----------
+    positionfixes : GeoDataFrame (as trackintel positionfixes)
+    staypoints : GeoDataFrame (as trackintel staypoints)
+    triplegs : GeoDataFrame (as trackintel triplegs)
+    locations : GeoDataFrame (as trackintel locations)
+    radius_sp : float
+    radius_locs : float
+    ax : matplotlib.pyplot.Artist
+    """
+    if positionfixes is not None:
+        positionfixes.plot(ax=ax, markersize=0.5)
+    if staypoints is not None:
+        center_latitude = (ax.get_ylim()[0] + ax.get_ylim()[1]) / 2
+        radius = meters_to_decimal_degrees(radius_sp, center_latitude)
+        geometry = staypoints.geometry.name
+        for pt in staypoints.to_dict("records"):
+            circle = mpatches.Circle(
+                (pt[geometry].x, pt[geometry].y), radius, facecolor="none", edgecolor="g", zorder=3
+            )
+            ax.add_artist(circle)
+    if triplegs is not None:
+        triplegs.plot(ax=ax, cmap="viridis")
+    if locations is not None:
+        center_latitude = (ax.get_ylim()[0] + ax.get_ylim()[1]) / 2
+        radius = meters_to_decimal_degrees(radius_locs, center_latitude)
+        for pt in locations.to_dict("records"):
+            circle = mpatches.Circle(
+                (pt["center"].x, pt["center"].y), radius, facecolor="none", edgecolor="r", zorder=4
+            )
+            ax.add_artist(circle)
+
+
+def plot(
+    *,
```

Comment:
Why the star?

Reply:
I wanted to enforce the keyword arguments. But yes that is maybe not necessary. I'll change it :)